controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -354,7 +354,7 @@ public OperationContext createContext(String scope, String name) {
 
 
     @Override
-    public CompletableFuture<List<Long>> getActiveSegments(final String scope, final String name, final long timestamp, final OperationContext context, final Executor executor) {
+    public CompletableFuture<Map<Long, Long>> getActiveSegments(final String scope, final String name, final long timestamp, final OperationContext context, final Executor executor) {
```

Comment:
no the client api remains same. it got segment/offset and it continues to do so. 
only difference is, instead of defaulting to `0` offset in controllerService, we do it in the store. And if there is a truncation record with higher offsets, then those offsets are taken instead of defaulting to 0.

Reply:
Ok.